troop/views.py

```diff
@@ -17,20 +20,13 @@
 class OnlyTroopManagerMixin(AccessMixin):
     """Verify that the current user is allowed to manage the troop."""
 
-    def dispatch(self, request, *args, **kwargs):
-        if not request.user.is_authenticated:
-            return self.handle_no_permission()
-
-        troop_number = kwargs.get("troop")
-        if not troop_number:
-            return self.handle_no_permission()
-
-        troop = request.user.troops.filter(number=troop_number).first()
+    def dispatch(self, request, troop_number=None, *args, **kwargs):
+        troop = Troop.managed_by_user(request.user, troop_number)
```

Comment:
Hm, I don't know. Wouldn't it be more beautiful if we either

a) get all troops a user can manage and check whether `troop_number` is part of these troops or
b) get the troop via `troop_number` and check whether this troop can be managed by the current user?

Reply:
I find it good enough like this.

Maybe we will see a better way when we have other usecases (this is currently used only in one place).